### PR TITLE
Replace cudf column.full with cudf.as_column

### PR DESCRIPTION
**Pull request by mroeschke**
## Description
I have a PR in cudf removing `full` in favor of `as_column` which has equivalent functionality (https://github.com/rapidsai/cudf/pull/14698), so replacing the single usage I found here

## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cuspatial/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

**Review by harrism — APPROVED** [review]

**Comment by vyasr** [comment]
@mroeschke looks like there is [one real error caused by RangeIndexes looking different](https://github.com/rapidsai/cuspatial/actions/runs/8073398134/job/22057284496?pr=1350#step:8:1057)

**Comment by harrism** [comment]
#1349 is now failing because of the failed import of `full`. I would like to get this merged. @mroeschke what is the blocker?

**Comment by bdice** [comment]
@mroeschke FYI, this PR is also blocking cuxfilter tests.

**Comment by bdice** [comment]
One legitimate failure is copied below (to separate from the noise of unrelated CI failures):
https://github.com/rapidsai/cuspatial/actions/runs/8086937898/job/22098408527?pr=1350#step:7:17280
``
=========================== short test summary info ============================
FAILED tests/spatial/distance/test_distance.py::test_geoseries_distance_indices_different - AssertionError: ColumnBase are different

values are different (100.0 %)
[left]:  {'[0.0, <NA>]'}
[right]: {'[0.0, nan]'}
```

File: python/cuspatial/cuspatial/core/binops/distance_dispatch.py [hunk 1]
```diff
@@ -185,10 +185,8 @@ def __call__(self):
 
         # Rows with misaligned indices contains nan. Here we scatter the
         # distance values to the correct indices.
-        result = full(
-            len(self._res_index),
-            float("nan"),
-            dtype="float64",
+        result = as_column(
+            [float("nan")] * len(self._res_index), dtype="float64", nan_as_null=False
         )
```
Comment (galipremsagar) [inline python/cuspatial/cuspatial/core/binops/distance_dispatch.py:190]:
```suggestion
        result = as_column(
            [float("nan")] * len(self._res_index),
            dtype="float64",
            nan_as_null=False,
        )
```

Comment (galipremsagar) [reply python/cuspatial/cuspatial/core/binops/distance_dispatch.py:190]: black failed, this should fix it

**Review by galipremsagar — APPROVED** [review]

**Review by bdice — APPROVED** [review]

**Comment by bdice** [comment]
I'll pre-emptively merge this since we expect things to pass and it's blocking other work.

File: python/cuspatial/cuspatial/core/binops/distance_dispatch.py [hunk 2]
```diff
@@ -185,10 +185,10 @@ def __call__(self):
 
         # Rows with misaligned indices contains nan. Here we scatter the
         # distance values to the correct indices.
-        result = full(
-            len(self._res_index),
-            float("nan"),
+        result = as_column(
+            [float("nan")] * len(self._res_index),
```
Comment (wence-) [inline python/cuspatial/cuspatial/core/binops/distance_dispatch.py:189]:
```suggestion
        result = as_column(
            float("nan"),
            length=len(self._res_index),
```

?

Comment (bdice) [reply python/cuspatial/cuspatial/core/binops/distance_dispatch.py:189]: That was tested in 4b6cb9f. See CI results:

```
FAILED python/cuspatial/cuspatial/tests/spatial/distance/test_distance.py::test_geoseires_distance_empty - AssertionError: Series are different

Series length are different
[left]:  1, RangeIndex(start=0, stop=1, step=1)
[right]: 0, RangeIndex(start=0, stop=0, step=1)
```

Comment (wence-) [reply python/cuspatial/cuspatial/core/binops/distance_dispatch.py:189]: With this diff against trunk:
```
diff --git a/python/cuspatial/cuspatial/core/binops/distance_dispatch.py b/python/cuspatial/cuspatial/core/binops/distance_dispatch.py
index c1995310..db2e6339 100644
--- a/python/cuspatial/cuspatial/core/binops/distance_dispatch.py
+++ b/python/cuspatial/cuspatial/core/binops/distance_dispatch.py
@@ -1,5 +1,5 @@
 import cudf
-from cudf.core.column import as_column, full
+from cudf.core.column import as_column
 
 from cuspatial._lib.distance import (
     pairwise_linestring_distance,
@@ -185,10 +185,11 @@ class DistanceDispatch:
 
         # Rows with misaligned indices contains nan. Here we scatter the
         # distance values to the correct indices.
-        result = full(
-            len(self._res_index),
+        result = as_column(
             float("nan"),
+            length=len(self._res_index),
             dtype="float64",
+            nan_as_null=False,
         )
         scatter_map = as_column(
             range(len(self._res_index)), dtype="int32"
```

I cannot replicate this error locally.

**Comment by mroeschke** [comment]
Looks like this PR is green except for the timed-out falures (I don't have permission to re-run those jobs)

**Comment by bdice** [comment]
I reran the failed jobs.

**Pull request merged**